src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -231,12 +231,33 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitGroupClause(
       SparqlAutomaticParser::GroupClauseContext* ctx) override {
-    return visitChildren(ctx);
+    vector<GroupKey> groupConditions;
+    for (auto* groupCondition : ctx->groupCondition()) {
+      groupConditions.push_back(
+          visitGroupCondition(groupCondition).as<GroupKey>());
+    }
+    return groupConditions;
   }
 
   antlrcpp::Any visitGroupCondition(
       SparqlAutomaticParser::GroupConditionContext* ctx) override {
-    return visitChildren(ctx);
+    if (ctx->var() && !ctx->expression()) {
+      return GroupKey{VariableGroupKey(ctx->var()->getText())};
+    } else if (ctx->builtInCall() || ctx->functionCall()) {
+      // builtInCall and functionCall are both also an Expression
+      auto expr = sparqlExpression::SparqlExpressionPimpl{
+          std::move(visit(ctx->builtInCall()).as<ExpressionPtr>())};
+      return GroupKey{ExpressionGroupKey{std::move(expr)}};
+    } else if (ctx->expression()) {
+      auto expr = sparqlExpression::SparqlExpressionPimpl{
```

Comment:
I think we have seen this Pattern so often, that it would be worth to create a helperFunction
` SparqlExpressionPimpl makeExpressionPimpl(antlrcpp::Any);`

Reply:
I found 5 places where this pattern was used. Seems a bit low to me.